### ISSUE TITLE
remove `id` from object after destroying it

### ISSUE THREAD

**Issue by vince1995** [parse-community/Parse-SDK-JS#1454]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases). 

### Issue Description
I've noticed that when you `destroy` an object, it still has the `id` property.

### Steps to reproduce
```
const obj = new Parse.Object("randomCollection");
await obj.save();
console.log(obj.id) // the id
await obj.destroy();
console.log(obj.id); // look at actual/expected outcome
```

### Actual Outcome

```
console.log(obj.id) // the id
```

### Expected Outcome

```
console.log(obj.id) // undefined
```

###  Environment

Server
- Parse Server version: `4.10.6`
- Operating system: `Ubuntu 20.04`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `local`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `5.0.3`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `local`

Client
- Parse JS SDK version: `3.4.1`

**Comment by mtrezza** [comment]
Can't follow your steps to reproduce. You are logging the obj ID after saving and before destroying. Why do you expect it be undefined?

**Comment by vince1995** [comment]
Because the object doesn't persist on the database anymore. So there's no record for this ID. So, why should the ID *not* be undefined?

**Comment by mtrezza** [comment]
I think you have left out the last step in steps to reproduce and added it instead to the "expected / actual outcome". Could you rewrite your example to make clear where you are expecting which outcome?

**Comment by vince1995** [comment]
Ah, yeah, sure. That wasn't clear.

**Comment by mtrezza** [comment]
I think I understand the issue description now. Could you explain why you expect `id` to be undefined? The object ID is vaguely indicative of the object state. If it's undefined it could also mean that the object has never been saved.

**Comment by vince1995** [comment]
Yeah, that's the point. In my opinien if it's undefined, it never persisted in the database or was deleted. But if the object ID is not undefined that should mean that this object has a record in the database.

So here's my question again: When an object is not in the database anymore, why should it have an object ID?

**Comment by mtrezza** [comment]
> So here's my question again: When an object is not in the database anymore, why should it have an object ID?

There are several aspects to consider:
- Inference
  The docs say:
  > objectId is a unique identifier for each saved object.
  
  That doesn't imply that the ID should be deleted after an object das been removed from the DB.
  In fact, none of the fields is unset after an object has been destroyed, see `createdAt`, `updatedAt`.
- Internal logic
  Does the ID have to be removed in order for the SDK to work properly?
  There may be internal logic in the Parse SDK or Parse Server that infers an object state from the presence of the ID field. If that was the case, this should be reviewed as a bug.

  For example this does not work:
  ```
  await obj.save(null, { useMasterKey: true});
  await obj.destroy();
  obj.unset("objectId");
  await obj.save(null, { useMasterKey: true}); // throws error "Object not found.", probably tries to update an object with the given ID
  ```
  Maybe it should be possible to easily save, delete and re-save an object, and assigning a new ID if custom obj ID is disabled.
- Developer expectation
  Do other Parse SDKs (or maybe even 3rd party SDKs) usually unset the object ID and is the Parse JS SDK an exception?
